libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/components/PickerBottomSheet.kt

```diff
@@ -0,0 +1,631 @@
+
+enum class PickerStyle {
+    LoginServerPicker,
+    UserAccountPicker,
+}
+
+internal const val PADDING_SIZE = 12
+internal const val ICON_SIZE = 32
+internal const val HEADER_TEXT_SIZE = 20
+internal const val HEADER_PADDING_SIZE = 16
+internal const val TEXT_SIZE = 16
+internal const val CORNER_RADIUS = 9
+internal const val STROKE_WIDTH = 1
+internal const val SLOW_ANIMATION_MS = 500
+internal const val TEXT_SELECTION_ALPHA = 0.2f
+
+@VisibleForTesting
+internal const val PICKER_CD = "Picker"
+@VisibleForTesting
+internal const val CLOSE_BUTTON_CD = "Close"
+@VisibleForTesting
+internal const val BACK_BUTTON_CD = "Back"
+@VisibleForTesting
+internal const val ADD_NEW_BUTTON_CD = "Add"
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PickerBottomSheet(pickerStyle: PickerStyle) {
+    val viewModel: LoginViewModel = viewModel(factory = SalesforceSDKManager.getInstance().loginViewModelFactory)
+    val loginServerManager = SalesforceSDKManager.getInstance().loginServerManager
+    val userAccountManager = SalesforceSDKManager.getInstance().userAccountManager
+    val activity = LocalContext.current.getActivity()
+    val backgroundColor = SalesforceSDKManager.getInstance().colorScheme().background
+    val onNewLoginServerSelected = { newSelectedServer: Any?, closePicker: Boolean ->
+        if (newSelectedServer != null && newSelectedServer is LoginServer) {
+            viewModel.showServerPicker.value = !closePicker
+            viewModel.loading.value = true
+            viewModel.dynamicBackgroundColor.value = backgroundColor
+            SalesforceSDKManager.getInstance().loginServerManager.selectedLoginServer = newSelectedServer
+        }
+    }
+    val onLoginServerCancel = { viewModel.showServerPicker.value = false }
+    val onUserAccountSelected = { userAccount: Any?, _: Boolean ->
+        if (userAccount != null && userAccount is UserAccount) {
+            activity?.finish()
+            userAccountManager.switchToUser(userAccount)
+        }
+    }
+    val onUserSwitchCancel = {
+        activity?.finish()
+        if (userAccountManager.currentUser == null) {
+            userAccountManager.switchToUser(userAccountManager.authenticatedUsers.first())
+        }
+    }
+    val sheetState = rememberModalBottomSheetState(
+        skipPartiallyExpanded = true,
+        confirmValueChange = { sheetValue ->
+            if (sheetValue == SheetValue.Hidden) {
+                when (pickerStyle) {
+                    PickerStyle.LoginServerPicker -> onLoginServerCancel()
+                    PickerStyle.UserAccountPicker -> onUserSwitchCancel()
+                }
+            }
+
+            true
+        }
+    )
+    val addNewLoginServer = { name: String, url: String ->
+        loginServerManager.addCustomLoginServer(name, url)
+        viewModel.showServerPicker.value = false
+        viewModel.loading.value = true
+        viewModel.dynamicBackgroundColor.value = backgroundColor
+    }
```

Comment:
These inline functions are the only **untested** part of the code.  My UI tests check that they are called but their actual contents can't be tested.  Maybe the Login Server ones should be added to `LoginViewModel`?  

The UserAccount ones wouldn't belong there and require reference to the activity (which should be avoided in view models IMO).